src/Phinx/Config/Config.php

```diff
@@ -154,6 +155,12 @@ public function getEnvironment($name)
                 $environments[$name]['default_migration_table'] =
                     $this->values['environments']['default_migration_table'];
             }
+			// inject any other custom $config entries outside of the $config['environments'] entry
+			foreach ($this->values as $key => $value) {
+				if ($key !== self::CONFIG_ENVIRONMENTS) {
+					$environments[$name][$key] = $this->values[$key];
```

Comment:
So if you have the following in phinx.php
$config = [
	'paths' => [
		'migrations' => 'migrations/company'
	],
	'environments' => [
		'default_migration_table' => 'phinx_migrations',
		'default_database' => 'none',
	],
	'tools' => $toolDefinitions,
];
It will now inject 'paths' and 'tools' into the $environments[$name] which has the basic information for each company.   What I had done previously was added 'tools' in below as follows
foreach ($databaseNames as $rowNumber => $databaseName) {
	$config['environments'][$databaseName['domain']] = [
		'adapter' => 'bamboohrmysql',
		'host' => $databaseName['db_host'],
		'name' => $databaseName['db_name'],
		'user' => $DB_USERNAME,
		'pass' => $DB_PASSWORD,
		'AES_ENCRYPTION_KEY' => $AES_ENCRYPTION_KEY,
		'company_id' => $databaseName['company_id'],
               'tools' => &$toolDefinitions,
	];
}
 which added significantly to the overhead of the $config as it was passed in to each migrate command